developing-without-a-build/article.md

```diff
@@ -0,0 +1,266 @@
+
+There a lot of different ways to write CSS, it's beyond the scope of this article to look into all of them. When working without a build, regular stylesheets willust work fine of course.
+
+If you are using some kind of CSS preprocessor you can run it before running your web server and just load the CSS output. Many CSS in JS solutions should also work if the library publishes an es module format.
+
+#### Shadow dom
+For truly modular CSS I recommend looking into [Shadow dom](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), it fixes many of the scoping and encapsulation problems of CSS. Unfortunately, it's not yet a complete story. There are still missing features that are being worked out in the standard so it may not yet be the [right solution in all scenarios](https://dev.to/webpadawan/beyond-the-polyfills-how-web-components-affect-us-today-3j0a).
+
+Good to mention here is [the lit-element library](https://www.npmjs.com/package/lit-element), which offers a great developer experience when authoring modular CSS without a build step. `lit-element` does most of the heavy lifting for you. You author CSS using tagged template literals, which are just syntax sugar for creating a [Constructable Stylesheet](https://developers.google.com/web/updates/2019/02/constructable-stylesheets). This way you can write and share CSS between your components.
+
+This will integrate well with CSS modules when they are shipped. We could emulate CSS modules by using fetch, but like we saw with HTML it's hard to optimize this for production use. I'm not a fan of CSS in JS, but I really feel that lit-element's solution is different and very elegant. You're writing CSS in a JS file, but you're still writing valid CSS syntax. If you like to keep things pure, you can just create a my-styles.css.js file and use a default export of just a stylesheet.
+
+### Library support
+Luckily the amount of libraries shipping as es module format is growing steadily. But there are still popular libraries which only as ship UMD or CommonJS. These don't work without some kind of code transformation. The best thing we can do it open issues on these projects to give them an indication how many people are interested in supporting the native module syntax.
+
+I think this is a problem that will disappear relatively quickly, especially after Node JS implements es modules as well. Many projects already use es modules as their authoring format, and I don't think anyone likes having to ship multiple module formats.
+
+## Building
+The goal of this article was to explore workflows where we don't need to do any building for development, and I think we've proven it's really possible. However it's important to reitarate that I don't think that build tools are evil, or that I want to pretend that noone will need to use any building at all during development.
```

Comment:
```suggestion
The goal of this article was to explore workflows where we don't need to do any building for development, and I think we've proven it's really possible. However it's important to reiterate that I don't think that build tools are evil, or that I want to pretend that no one will need to use any building at all during development.
```

Reply:
This got outdated, so I can't apply the changes anymore. Applied it manually, thanks!